langkit/passes.py

```diff
@@ -92,9 +92,31 @@ class AbstractPass:
     disable passes in big PassManager.add calls.
     """
 
+    is_optional: bool
+    """
+    If true, the pass can be activated/deactivated via the command line
+    arguments.
+    """
+
+    doc: str
+
     def __init__(self, name: str, disabled: bool = False) -> None:
         self.name = name
         self.disabled = disabled
+        self.is_optional = False
+
+    def optional(self, doc: str, disabled: bool = True) -> AbstractPass:
```

Comment:
Given that this does not create a new pass (which we would not need anyway), what about just adding doc/optional arguments to the constructor? Then for instance have `disabled: Optional[bool] = None`, which results in `disabled = True` if optional and `disabled = False` otherwise.

Reply:
I didn't do that because it requires much more boilerplate and refactoring:

* Adding the `optional` argument to every constructor that requires it (so potentially every subclass)
* Forwarding the arguments to the super call

For not much practical benefits

Reply:
Ok. I still prefer the constructor arg (as I did for `disabled`), as I find the uses of `.optional` awkward, but I have no problem considering this as a matter of taste, so I won’t object.

Reply:
I forgot to add: regarding docstring enforcement/runtime check: I guess we could enforce doc for optional passes in constructor with Mypy’s overloads, but that does not change my conclusion (I won’t object).

Reply:
Ok